src/server/api/controllers/orders.controller.js

```diff
@@ -0,0 +1,56 @@
+const knex = require('../../config/db');
+const HttpError = require('../lib/utils/http-error');
+const moment = require('moment-timezone');
+
+
+
+const newOrder = async({ user_id, items }) => {
+    const userId = parseInt(user_id, 10);
+    if (isNaN(userId) || userId < 1) {
+        throw new HttpError(
+            'Bad request. User ID must be an integer and larger than 0',
+            400,
+        );
+    }
+    try {
+        const orderId = await knex('orders')
+            .insert({
+                user_id,
+                created_at: moment(Date.now()).format(),
+                status: 'NEW',
+            })
+            .returning('id');
+
+        console.log({ orderId })
+
+        items.forEach(async(item) => {
+            console.log({ item })
```

Comment:
If you remove `console.log` this will break again. `console.log` is odd because it blocks the event loop for a while which allows Promises to complete in the background.

Please check my comment on the last version to see how to do this properly.

Reply:
Hej Milton. Thank you for your feedback. I tested your code before but it gives me the internal server error. Please see the screenshots below. I'm not sure how to fix it.

![Screenshot 2022-02-28 at 20 30 41](https://user-images.githubusercontent.com/81103031/156048940-50295a51-422e-47ca-b9d0-b20173290905.png)
![Screenshot 2022-02-28 at 20 29 47](https://user-images.githubusercontent.com/81103031/156048951-cb2ac970-bb6e-4950-848e-88c1cbdf194e.png)
.

Reply:
First of you need to understand what the Internal server error is caused by. 
Before throwing `HttpError 500` you can use `console.log` to see what is the cause.

Reply:
If I compare the code you wrote there is uses `orderId` but the PR code uses `orderId[0]`, could that be the issue?